metrics-operator/controllers/common/providers/prometheus/prometheus.go

```diff
@@ -30,19 +30,52 @@ func (r *KeptnPrometheusProvider) EvaluateQuery(ctx context.Context, metric metr
 		return "", nil, err
 	}
 	api := prometheus.NewAPI(client)
-	result, w, err := api.Query(
-		ctx,
-		metric.Spec.Query,
-		queryTime,
-		[]prometheus.Option{}...,
-	)
+	var result model.Value
+	var warnings prometheus.Warnings
+	
+	if metric.Spec.Range != nil {
 
-	if err != nil {
-		return "", nil, err
-	}
+		queryInterval, err := time.ParseDuration(metric.Spec.Range.Interval)
+		if err != nil {
+			return "", nil, err
+		}
+		endTime := queryTime.Add(queryInterval)
+
+		queryRange := prometheus.Range{
+			Start: queryTime,
+			End: endTime,
+		}
 
-	if len(w) != 0 {
-		r.Log.Info("Prometheus API returned warnings: " + w[0])
+		r, w, err := api.QueryRange(
```

Comment:
please rename the return value `r` to something else, as this shadows the declaration of the pointer receiver